tests/functions/module_array.test.ts

```diff
@@ -448,6 +448,55 @@ describe("FREQUENCY function", () => {
   });
 });
 
+describe("SUMPRODUCT function", () => {
+  test("SUMPRODUCT takes at least 1 argument", () => {
+    expect(evaluateCell("A1", { A1: "=SUMPRODUCT()" })).toBe("#BAD_EXPR"); // @compatibility: on google sheets, return #N/A
+    expect(evaluateCell("A1", { A1: "=SUMPRODUCT(5)" })).toBe(5);
+    expect(evaluateCell("A1", { A1: "=SUMPRODUCT(5, 5)" })).toBe(25);
+  });
+
+  test("Values must be numbers, or range of numbers", () => {
+    expect(evaluateCell("A1", { A1: '=SUMPRODUCT("hallo")' })).toBe("#ERROR"); // @compatibility: on google sheets, return #VALUE!
+    expect(evaluateCell("A1", { A1: "=SUMPRODUCT(B1:B2, B1:B2)", B1: "hey", B2: "yo" })).toBe(
+      "#ERROR"
+    ); // @compatibility: on google sheets, return #VALUE!
+  });
+
+  test("Range values must have the same dimensions", () => {
+    expect(evaluateCell("D1", { D1: "=SUMPRODUCT(A1:A2, A1:B1)" })).toBe("#ERROR"); // @compatibility: on google sheets, return #VALUE!
+    expect(evaluateCell("D1", { D1: "=SUMPRODUCT(A1:A2, A1:A3)" })).toBe("#ERROR"); // @compatibility: on google sheets, return #VALUE!
+  });
+
+  test("SUMPRODUCT with numbers arguments", () => {
```

Comment:
what happens if number arguments are mixed with ranges?

Reply:
Was supposed to fail like in the test `Range values must have the same dimensions` , but the helper `assertSameDimensions` had an error. Fixed it and added a test.